### PR TITLE
Updated the model card for CLIP

### PR DESCRIPTION
**Pull request by purusharthmalik**
# What does this PR do?

As suggested in this issue - https://github.com/huggingface/transformers/issues/36979#issue-2947704577 - this PR updates the documentation of the CLIP model, which will now be aligned with the standardized format for all the docs.

## Who can review?

@stevhliu, please let me know if any changes are needed.

### CODE REVIEW

**Comment by purusharthmalik** [comment]
@stevhliu Do I need to make any changes?

File: docs/source/en/model_doc/clip.md [hunk 1]
```diff
@@ -14,221 +14,149 @@ rendered properly in your Markdown viewer.
...
 
->>> inputs = processor(text=["a photo of a cat", "a photo of a dog"], images=image, return_tensors="pt", padding=True)
+The example below demonstrates how to calculate similarity scores between multiple textual options and a given image with [`Pipeline`], [`AutoModel`], and [`CLIPModel`].
 
->>> outputs = model(**inputs)
->>> logits_per_image = outputs.logits_per_image  # this is the image-text similarity score
->>> probs = logits_per_image.softmax(dim=1)  # we can take the softmax to get the label probabilities
-```
+<hfoptions id="usage">
+<hfoption id="Pipeline">
```
Comment (stevhliu) [inline docs/source/en/model_doc/clip.md:40]: The Pipeline snippet should look like:

```py
import torch
from transformers import pipeline

clip = pipeline(
   task="zero-shot-image-classification",
   model="openai/clip-vit-base-patch32",
   torch_dtype=torch.bfloat16,
   device=0
)
labels = ["a photo of a cat", "a photo of a dog", "a photo of a car"]
clip("http://images.cocodataset.org/val2017/000000039769.jpg", candidate_labels=labels)
```

File: docs/source/en/model_doc/clip.md [hunk 2]
```diff
@@ -14,221 +14,149 @@ rendered properly in your Markdown viewer.
...
-</Tip>
-
-To load and run a model using Flash Attention 2, refer to the snippet below:
-
-```python
->>> import torch
->>> import requests
->>> from PIL import Image
+</hfoption>
+<hfoption id="AutoModel">
```
Comment (stevhliu) [inline docs/source/en/model_doc/clip.md:58]: The AutoModel snippet should look like:

```py
import requests
import torch
from PIL import Image
from transformers import AutoProcessor, AutoModel

model = AutoModel.from_pretrained("openai/clip-vit-base-patch32", torch_dtype=torch.bfloat16, attn_implementation="sdpa")
processor = AutoProcessor.from_pretrained("openai/clip-vit-base-patch32")

url = "http://images.cocodataset.org/val2017/000000039769.jpg"
image = Image.open(requests.get(url, stream=True).raw)
labels = ["a photo of a cat", "a photo of a dog", "a photo of a car"]

inputs = processor(text=labels, images=image, return_tensors="pt", padding=True)

outputs = model(**inputs)
logits_per_image = outputs.logits_per_image
probs = logits_per_image.softmax(dim=1)
most_likely_idx = probs.argmax(dim=1).item()
most_likely_label = labels[most_likely_idx]
print(f"Most likely label: {most_likely_label} with probability: {probs[0][most_likely_idx].item():.3f}")
```

File: docs/source/en/model_doc/clip.md [hunk 3]
```diff
@@ -14,221 +14,149 @@ rendered properly in your Markdown viewer.
...
->>> logits_per_image = outputs.logits_per_image  # this is the image-text similarity score
->>> probs = logits_per_image.softmax(dim=1)  # we can take the softmax to get the label probabilities
->>> print(probs)
-tensor([[0.9946, 0.0052]], device='cuda:0', dtype=torch.float16)
+similarity = torch.cosine_similarity(image_features, text_features)
+print(similarity)
 ```
 
+</hfoption>
+<hfoption id="CLIPModel">
```
Comment (stevhliu) [inline docs/source/en/model_doc/clip.md:85]: Let's remove the `CLIPModel` so users can just focus on `Pipeline` and `AutoModel`

File: docs/source/en/model_doc/clip.md [hunk 4]
```diff
@@ -14,221 +14,149 @@ rendered properly in your Markdown viewer.
...
-|                 4 |            0.009 |          0.012 |         0.737 |           0.007 |          1.269 |
-|                16 |            0.009 |          0.014 |         0.659 |           0.008 |          1.187 |
-|                32 |            0.018 |          0.021 |         0.862 |           0.016 |          1.142 |
-|                64 |            0.034 |          0.034 |         1.001 |           0.03  |          1.163 |
-|               128 |            0.063 |          0.058 |         1.09  |           0.054 |          1.174 |
+</hfoption>
+</hfoptions>
 
-![clip_text_model_viz_3](https://github.com/user-attachments/assets/e9826b43-4e66-4f4c-952b-af4d90bd38eb)
+## Using CLIP with Quantization and Flash Attention 2
```
Comment (stevhliu) [inline docs/source/en/model_doc/clip.md:109]: We can also remove the rest of this to keep the focus on just `Pipeline` and `AutoModel` (which already uses sdpa).

File: docs/source/en/model_doc/clip.md [hunk 5]
```diff
@@ -14,221 +14,149 @@ rendered properly in your Markdown viewer.
...
-**Explainability**
+similarity = torch.cosine_similarity(image_features, text_features)
+```
 
-- A [notebook](https://colab.research.google.com/github/hila-chefer/Transformer-MM-Explainability/blob/main/CLIP_explainability.ipynb) on how to visualize similarity between input token and image segment. 🌎
+## Notes
 
-If you're interested in submitting a resource to be included here, please feel free to open a Pull Request and we will review it.
-The resource should ideally demonstrate something new instead of duplicating an existing resource.
+- For the best speedups, we recommend loading the model in half-precision (e.g. torch.float16 or torch.bfloat16).
```
Comment (stevhliu) [inline docs/source/en/model_doc/clip.md:159]: Not necessary at the moment since we show how to load in half-precision by default

**Review by stevhliu — COMMENTED** [review]
Fantastic start! 🔥

**Comment by purusharthmalik** [comment]
All the suggested changes have been incorporated into the docs. Also, are model-specific loaders and processors being deprecated or are we just omitting them from the initial description about the model for standardization's sake?

File: docs/source/en/model_doc/clip.md [hunk 6]
```diff
@@ -14,221 +14,78 @@ rendered properly in your Markdown viewer.
...
-To load and run a model using Flash Attention 2, refer to the snippet below:
-
-```python
->>> import torch
->>> import requests
->>> from PIL import Image
+# CLIP
 
->>> from transformers import CLIPProcessor, CLIPModel
+[CLIP](https://huggingface.co/papers/2103.00020) is a is a multimodal vision and language model motivated by overcoming the fixed number of object categories when training a computer vision model. CLIP learns about images directly from raw text by jointly training on 400M (image, text) pairs with an image encoder and text encoder. Both features are projected to a latent space with the same number of dimensions and their dot product gives a similarity score.
```
Comment (stevhliu) [inline docs/source/en/model_doc/clip.md:30]:
```suggestion
[CLIP](https://huggingface.co/papers/2103.00020) is a is a multimodal vision and language model motivated by overcoming the fixed number of object categories when training a computer vision model. CLIP learns about images directly from raw text by jointly training on 400M (image, text) pairs. Pretraining on this scale enables zero-shot transfer to downstream tasks. CLIP uses an image encoder and text encoder to get visual features and text features. Both features are projected to a latent space with the same number of dimensions and their dot product gives a similarity score.
```

File: docs/source/en/model_doc/clip.md [hunk 7]
```diff
@@ -14,221 +14,78 @@ rendered properly in your Markdown viewer.
...
+</hfoption>
+</hfoptions>
 
-- A [notebook](https://colab.research.google.com/github/hila-chefer/Transformer-MM-Explainability/blob/main/CLIP_explainability.ipynb) on how to visualize similarity between input token and image segment. 🌎
+## Notes
 
-If you're interested in submitting a resource to be included here, please feel free to open a Pull Request and we will review it.
-The resource should ideally demonstrate something new instead of duplicating an existing resource.
+- If you're interested in submitting a resource to be included here, please feel free to open a Pull Request and we will review it.
+- The resource should ideally demonstrate something new instead of duplicating an existing resource.
```
Comment (stevhliu) [inline docs/source/en/model_doc/clip.md:88]:
```suggestion
- Use [`CLIPImageProcessor`] to resize (or rescale) and normalizes images for the model.
```

File: docs/source/en/model_doc/clip.md [hunk 8]
```diff
@@ -14,221 +14,78 @@ rendered properly in your Markdown viewer.
...
->>> import requests
->>> from PIL import Image
+# CLIP
 
->>> from transformers import CLIPProcessor, CLIPModel
+[CLIP](https://huggingface.co/papers/2103.00020) is a is a multimodal vision and language model motivated by overcoming the fixed number of object categories when training a computer vision model. CLIP learns about images directly from raw text by jointly training on 400M (image, text) pairs with an image encoder and text encoder. Both features are projected to a latent space with the same number of dimensions and their dot product gives a similarity score.
 
->>> device = "cuda"
->>> torch_dtype = torch.float16
+You can find all the original CLIP checkpoints under the [OpenAI](https://huggingface.co/openai) organization.
```
Comment (stevhliu) [inline docs/source/en/model_doc/clip.md:32]:
```suggestion
You can find all the original CLIP checkpoints under the [OpenAI](https://huggingface.co/openai?search_models=clip) organization.
```

**Review by stevhliu — APPROVED** [review]
Super nice! Just a few more changes and we can merge 🚀

**Comment by purusharthmalik** [comment]
@stevhliu Done! Let's merge this, I'll get started on another model.
